### PR TITLE
Smooth Wall Fix

### LINKED ISSUE #12896: Standard and reinforced walls don't visually connect with each other anymore.
They used to link together, but at the moment they'll appear separately, like this.

image

### PR DESCRIPTION
**Pull request by 1Sparky1**
Fixes #12896

### CODE REVIEW

**Comment by 1Sparky1** [comment]
!bugfix

**Comment by 1Sparky1** [comment]
!review

**Review by alsoandanswer — APPROVED** [review]

File: code/game/turfs/simulated/walls.dm [hunk 1]
```diff
@@ -46,6 +46,10 @@
 			/obj/structure/window/full,
 			/obj/structure/window/full/phoron
 		)
+		if(istype(src, /turf/simulated/wall/r_wall))
+			LAZYADD(canSmoothWith, /turf/simulated/wall)
+		if(src.type == /turf/simulated/wall)
+			LAZYADD(canSmoothWith, /turf/simulated/wall/r_wall)
```
Comment (Geevies) [inline code/game/turfs/simulated/walls.dm:52]: this should probably be a proc on `/turf/simulated/wall`. something like `/turf/simulated/wall/proc/select_wall_smooth()` which gets overridden to add the other wall type. doing istype(src) is never good

**Review by Geevies — CHANGES_REQUESTED** [review]

**Review by Geevies — APPROVED** [review]

**Comment by NonQueueingMatt** [comment]
I don't particularly like the type hardchecking. Ideally this is handled in another way.

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**